### PR TITLE
Add HelixAttempt support to helix job sending

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Optionally read and pass the HelixAttempt variable through to the job start call

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
@maririos @alexperovich @mmitche I think this is the only change in BuildTools to get this sent through. There will be changes in corefx and coreclr to set HelixAttempt to OfficialBuildId if ProdConBuildId is set, but that's a bit later.

**Comment by maririos** [comment]
We might want to add documentation for the `Attempt` variable in https://github.com/ChadNedzlek/buildtools/blob/ccf477726ca4afdebc6ef5710d0d01a3db03e8e5/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets#L54

**Comment by ChadNedzlek** [comment]
@maririos Ooo, I missed that documentation part. Thanks. I updated it. It's a little weird that this documentation is one of several copies of this (the API itself and then some random .md files floating around), but I added a brief description.

**Comment by maririos** [comment]
That is true, documentation everywhere that gets out of date so fast :(
I'm not sure where else we explain what you could send to a Job but it is worth to adding it everywhere.

**Comment by ChadNedzlek** [comment]
Hrm... I want to wait until we roll out to prod for this one, so we don't accidentally get someone passing the new value and getting a 400 from the API.

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets [hunk 1]
```diff
@@ -51,6 +51,8 @@
        HelixArchLabel    - If HelixJobProperties is not set, we'll use this to fill it out    Default: <Blank>
        HelixConfigLabel  - If HelixJobProperties is not set, we'll use this to fill it out    Default: <Blank>
        MaxRetryCount     - Max automatic retry of workitems which do not return 0             Default: 0 (no retry)
+       HelixAttempt      - A lexically monotonic increasing string distiquishing              Default: <Blank>
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets]: distinguishing

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets [hunk 2]
```diff
@@ -51,7 +51,7 @@
        HelixArchLabel    - If HelixJobProperties is not set, we'll use this to fill it out    Default: <Blank>
        HelixConfigLabel  - If HelixJobProperties is not set, we'll use this to fill it out    Default: <Blank>
        MaxRetryCount     - Max automatic retry of workitems which do not return 0             Default: 0 (no retry)
-       HelixAttempt      - A lexically monotonic increasing string distiquishing              Default: <Blank>
+       HelixAttempt      - A lexically monotonic increasing string distinquishing             Default: <Blank>
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets]: distinGuishing :)

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets]: didn't even notice my g didn't have a little tail...

**Pull request merged**